mldatafind/authenticate.py

```diff
@@ -14,18 +16,49 @@ def authenticate():
     it will create a new one. If the credential exists and is valid,
     it will continue to use it. Otherwise, it will generate a new credential.
 
-    This function assumes the user has already made kerberos credentials.
+    If generating new credential, a kerberos keytab is required
+    for passwordless authentication. It's location should be
+    specified in the environment variable `KRB5_KTNAME`.
+    This function assumes the user has already generated a kerberos keytab
+    with principal user.name@LIGO.ORG. This function will read in username
+    from the environment variable `LIGO_USERNAME`
 
-    To generate kerberos credentials, run
-
-    ```kinit albert.einstein@LIGO.ORG```
+    For instructions on generating a kerberos keytab,
+    see https://computing.docs.ligo.org/guide/auth/kerberos/
 
     """
 
+    user = os.environ.get("LIGO_USERNAME")
+    if user is None:
+        raise ValueError("LIGO_USERNAME environment variable is not set")
+
+    keytab_location = os.getenv("KRB5_KTNAME")
+    if keytab_location is None:
+        raise ValueError("KRB5_KTNAME environment variable not set")
+
+    path = os.getenv("X509_USER_PROXY")
+    if path is None:
+        raise ValueError("X509_USER_PROXY environment variable not set")
+
+    kinit_command = shutil.which("kinit")
+    if kinit_command is None:
+        raise ValueError("kinit command not found")
+
+    kinit_command = shutil.which("kinit")
+    args = [
+        kinit_command,
+        "-p",
+        f"{user}@LIGO.ORG",
+        "-k",
+        "-t",
+        keytab_location,
+    ]
+    subprocess.run(args, check=True)
+
     path = os.getenv("X509_USER_PROXY")
 
     if path is None:
```

Comment:
Will doing the `kinit` possibly unset or change this variable?

Reply:
As far as i'm aware no, but should keep in mind.